spec/facts_spec.rb

```diff
@@ -41,22 +41,22 @@
 end
 
 describe 'Default Facts' do
-  before(:each) do
+  before do
     ENV['FACTERDB_SKIP_DEFAULTDB'] = nil
     FacterDB.instance_variable_set(:@database, nil)
   end
 
   describe 'fact files' do
-    it 'should not contain duplicate fact sets' do
+    it 'does not contain duplicate fact sets' do
       # Gather all of the default files and hash the content
       file_hashes = {}
       FacterDB.default_fact_files.each do |filepath|
-        file_hash = Digest::SHA256.hexdigest( File.open(filepath, 'rb') { |file| file.read } )
+        file_hash = Digest::SHA256.hexdigest(File.open(filepath, 'rb', &:read))
```

Comment:
Isn't this even shorter and the same?
```suggestion
        file_hash = Digest::SHA256.hexdigest(File.read(filepath, 'rb'))
```

Reply:
Eh? I am at a loss...

```
     Failure/Error: file_hash = Digest::SHA256.hexdigest(File.read(filepath, 'rb'))
     
     TypeError:
       no implicit conversion of String into Integer
```
```
     Failure/Error: file_hash = Digest::SHA256.hexdigest(File.read(filepath, 'r'))
     
     TypeError:
       no implicit conversion of String into Integer
```

```
     Failure/Error: file_hash = Digest::SHA256.hexdigest(File.read(filepath, 'r:iso-8859-1'))
     
     TypeError:
       no implicit conversion of String into Integer
```

```
     Failure/Error: file_hash = Digest::SHA256.hexdigest(File.read(filepath, 'rt'))
     
     TypeError:
       no implicit conversion of String into Integer
```

Reply:
It looks like you shouldn't pass the mode that way:
```ruby
[6] pry(main)> File.read('/etc/passwd', 'rb')
TypeError: no implicit conversion of String into Integer
```

I'm not sure you need to.

Reply:
`file_hash = Digest::SHA256.hexdigest(File.read(filepath))` avoids the conversion error.  I'm guessing that #read doesn't accept the options we think it should.  The docs don't even list `File#read` as a class method. https://ruby-doc.org/core-2.7.0/File.html#method-c-open  I don't understand where it is coming from... https://github.com/ruby/ruby/blob/master/file.c

Reply:
I think `File#binread` is probably what we were looking for...

Reply:
Converted to `File.binread`.

Reply:
I believe it's inherited from [IO](https://ruby-doc.org/core-3.1.0/IO.html) which defines these methods.

Reply:
I thought something magical was going on because I was sure that IO.read doesn't slurp unopened files... turns out that it will slurp, at least in my testing.